src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -376,10 +376,20 @@ public bool TryGetVariableType(string name, out FormulaType type)
 
             return false;
         }
-
+        
         public void UpdateSupportedFunctions(SymbolTable s)
         {
             SupportedFunctions = s;
         }
+        
+        public void AddNamedFormulas(string script, CultureInfo parseCulture = null)

```

Comment:
IIRC, doesn't RecalcEngine have a culture/parser options set already? Or should this take a ParserOptions like Eval appears to.

Reply:
`Engine.Check()` takes a `ParserOptions`. IMO its better to pass only the `CultureInfo` since side effects are not allowed in UDF/NF.